src/test/py/bazel/bzlmod/bazel_module_test.py

```diff
@@ -338,10 +338,10 @@ def testCheckDirectDependencies(self):
     self.AssertExitCode(exit_code, 48, stderr)
     stderr = '\n'.join(stderr)
     self.assertIn(
-        'ERROR: For repository \'aaa\', the root module requires module version aaa@1.0, but got aaa@1.1 in the resolved dependency graph.',
+        'ERROR: For repository \'aaa\', the root module requires module version aaa@1.0, but got aaa@1.1 in the resolved dependency graph. Source module(s): bbb@1.1, ccc@1.1',
         stderr)
     self.assertIn(
-        'ERROR: For repository \'bbb\', the root module requires module version bbb@1.0, but got bbb@1.1 in the resolved dependency graph.',
+        'ERROR: For repository \'bbb\', the root module requires module version bbb@1.0, but got bbb@1.1 in the resolved dependency graph. Source module(s): ccc@1.1',
```

Comment:
If I understand correctly, the "source modules" listed here may actually not be direct dependency of the root module? If so, the message could still be confusing.

I think a better message would be something showing a path to the required module, like `Introduced by: ccc@1.1 -> bbb@1.1 -> aaa@a.1`?